backend/Dockerfile

```diff
@@ -13,32 +26,40 @@ RUN apt update && apt install -y \
     gcc \
     && rm -rf /var/lib/apt/lists/*
 
-# install python libraries (except torch)
-COPY requirements/ requirements/
 ENV PIP_CERT=$CACERT_LOCATION
-RUN pip --default-timeout=300 install --upgrade pip \
-    && pip --default-timeout=300 install --no-cache-dir -r requirements/common.txt \
-    && rm -r /root/.cache
+
+# Download dependencies as a separate step to take advantage of Docker's caching.
+# Leverage a cache mount to /root/.cache/pip to speed up subsequent builds.
+# Leverage a bind mount to requirements.txt to avoid having to copy them into
+# into this layer.
+RUN --mount=type=cache,target=/root/.cache/pip \
+    --mount=type=bind,source=requirements.txt,target=requirements.txt \
+    python -m pip install -r requirements.txt
 
 ARG VERSION
 ARG MODEL="EffB7_2023-03-06_08"
 ENV SSL_CERT_FILE=$CACERT_LOCATION
 RUN curl -o model.pth https://storage.gra.cloud.ovh.net/v1/AUTH_df731a99a3264215b973b3dee70a57af/basegun-public/models/${MODEL}/${MODEL}.pth
-COPY src/ src/
+COPY . .

```

Comment:
here you copy unecessary stuff like the tests/, why not leave src/ ?

Reply:
To follow the "standard" Dockerfile. You want to have, as much as possible, the same docker image in each environment. We should test the same image that we are going to deploy, not a different one with or without the tests.